### PR TITLE
Deui reducer

### PR DESCRIPTION
**Pull request by 121watts**
- [x] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### The problem
Query configs not scoped to data explorer.

### The Solution
Create a reducer for DE state, and handle scoping of configs there.

### CODE REVIEW

File: ui/src/data_explorer/reducers/timeRange.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
       const {upper, lower} = action.payload;
+      const newState = {
+        upper,
+        lower,
+      };
 
-      return update(state, {
-        ['lower']: {$set: lower},
-        ['upper']: {$set: upper},
-      });
```
Comment (cryptoquick) [inline ui/src/data_explorer/reducers/timeRange.js]: for shame 😞

File: ui/src/data_explorer/reducers/timeRange.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+      return {...state, ...newState};
```
Comment (cryptoquick) [inline ui/src/data_explorer/reducers/timeRange.js]: 👍

File: ui/src/data_explorer/reducers/ui.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      return {...state, ...newState};
+    }
+
+    case 'DELETE_QUERY': {
+      const {queryID} = action.payload;
+      const newState = {
+        queryIDs: state.queryIDs.filter(id => id !== queryID),
+      };
+
+      return {...state, ...newState};
```
Comment (cryptoquick) [inline ui/src/data_explorer/reducers/ui.js]: Does line 22 do anything? Am I wrong in thinking this should be simply newState, or is there something I'm missing?

Comment (cryptoquick) [inline ui/src/data_explorer/reducers/ui.js]: I feel like this should be reaching into the object and doing something with `queryIDs`

File: ui/src/localStorage.js [hunk 4]
```diff
@@ -1,29 +1,21 @@
...
-
-    if (serializedState === null) {
-      return {timeRange: defaultTimeRange};
-    }
-
     const parsedState = JSON.parse(serializedState) || {};
-    const timeRange = _.isEmpty(parsedState.timeRange) ? defaultTimeRange : parsedState.timeRange;
 
-    return {...parsedState, timeRange};
+    return {...parsedState};
```
Comment (cryptoquick) [inline ui/src/localStorage.js]: This doesn't need to be unrolled, right? It's going to be a new immutable object either way, right?

Comment (121watts) [inline ui/src/localStorage.js]: That's totally correct.  Was to distracted by deleting all the things.

**Pull request merged**